### ISSUE TITLE
Merge reactions with original message

### ISSUE THREAD

**Issue by gferon** [whisperfish/presage#123]
Relating to #95, when we receive a reaction, we could either:
- modify the original stored message
- save the reaction without a reference to the original message
and send back the result in the stream.

This will help apps not having to merge those events themselves when rendering.

**Comment by Schmiddiii** [comment]
I agree that this would be nice to have. Regarding the possible options:

> modify the original stored message

I don't really think that would be nice, as the `reaction`-field for messages is for reacting for other messages and should probably not be abused for stroring reactions to the stored message. (Unless we want to introduce a completely new Message-type wrapping the `DataMessage` that will be stored).

> store the reactions aside somewhere

I agree with that option the most. Maybe introduce a new tree for every thread mapping the message timestamp to a list of reactions or only one tree mapping a combination of thread and message timestamp to the list of reactions.

> save the reaction without a reference to the original message and send back the result in the stream

Not sure what you exactly mean by that.
